gateway/ln-gateway/src/lib.rs

```diff
@@ -496,6 +509,7 @@ impl Gateway {
                         continue;
                     }
                 };
+
             if !route_hints.is_empty() || num_retries == ROUTE_HINT_RETRIES {
```

Comment:
Can we either:

1. Add a condition for `num_route_hints == 0` here to also exit in that case because then we expect the `route_hints` to be empty?
2. Add an early return path before the loop if `num_route_hints == 0` because we don't even need to fetch the routes in that case (although that might mask RPC connectivity problems and we should test that separately).

Reply:
This should be handled at line 510:

```
if num_route_hints == 0 {
            return Ok(vec![]);
        }
```

Reply:
Sorry, I'm an idiot and didn't realize I was in per-commit view (and just looked at the first one apparently).